app/service/search_index.rb

```diff
@@ -9,35 +9,57 @@ def self.log_elasticsearch_warning(message)
     end
   end
 
-  def initialize(model_name:, id:)
+  def initialize(model_name:, id:, changed: nil)
     @model_name = model_name
     @id = id
+    @changed = changed
   end
 
   def add
-    if search_index_callbacks_enabled?
-      record = model_name.constantize.find(id)
-      record.__elasticsearch__.index_document
-    else
-      log_callback_warning
+    perform { record.__elasticsearch__.index_document }
+  end
+
+  def update
+    return add unless changed.any?
+
+    perform do
+      data = elasticsearch_hash.merge(body: { doc: changes })
+      Elasticsearch::Model.client.update(data)
```

Comment:
Okay, I remember now. It wasn't working because we use Sidekiq and load the model from the database in the `SearchIndex` instead of using the loaded model on which the changes were made, therefore it doesn't have the changes list available for `update_document`. That's why instead I use the ES Client and pass it the changes directly through Sidekiq.